test/extensions/filters/http/common/factory_base_test.cc

```diff
@@ -0,0 +1,58 @@
+#include "envoy/server/filter_config.h"
+
+#include "common/protobuf/message_validator_impl.h"
+
+#include "extensions/filters/http/common/factory_base.h"
+
+#include "test/extensions/filters/http/common/factory_base_test.pb.validate.h"
+#include "test/mocks/server/mocks.h"
+#include "test/test_common/test_runtime.h"
+
+#include "gmock/gmock.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Common {
+namespace {
+
+class FakeFilterFactory
+    : public FactoryBase<test::extensions::filters::http::common::FakeFilterConfig> {
+public:
+  FakeFilterFactory() : FactoryBase("test") {}
+
+private:
+  Http::FilterFactoryCb createFilterFactoryFromProtoTyped(
+      const test::extensions::filters::http::common::FakeFilterConfig&, const std::string&,
+      Server::Configuration::FactoryContext&) override {
+    return nullptr;
+  }
+};
+
+TEST(FactoryBaseTest, TestRejectUnsupportedTypedFilterConfig) {
+  TestScopedRuntime scoped_runtime;
+  FakeFilterFactory factory;
+  NiceMock<Server::Configuration::MockServerFactoryContext> context;
+  ProtobufTypes::MessagePtr config = factory.createEmptyRouteConfigProto();
+
+  Runtime::LoaderSingleton::getExisting()->mergeValues(
+      {{"envoy.reloadable_features.check_unsupported_typed_per_filter_config", "true"}});
+  EXPECT_THROW(factory.createRouteSpecificFilterConfig(*config, context,
+                                                       ProtobufMessage::getNullValidationVisitor()),
+               EnvoyException);
+}
+
+TEST(FactoryBaseTest, TestAcceptUnsupportedTypedFilterConfig) {
+  FakeFilterFactory factory;
+  NiceMock<Server::Configuration::MockServerFactoryContext> context;
+  ProtobufTypes::MessagePtr config = factory.createEmptyRouteConfigProto();
+
+  EXPECT_EQ(nullptr, factory.createRouteSpecificFilterConfig(
+                         *config, context, ProtobufMessage::getNullValidationVisitor()));
```

Comment:
It would be good to have some e2e integration tests that cover this functionality.  I don't know exactly what existing types would fail config load if you tried to provide per-route config for them.  It seems that Envoy::Extensions::HttpFilters::HealthCheck::HealthCheckFilterConfig doesn't implement an override for createRouteSpecificFilterConfigTyped, perhaps that is a good extension to write a test against?

Reply:
Yea, `HealthCheckFilterConfig` doesn't implement override, let me add integration tests.